crates/aptos-rest-client/src/lib.rs

```diff
@@ -234,6 +235,58 @@ impl Client {
         resp.and_then(|resource| Ok(resource.coin()))
     }
 
+    pub async fn view_account_balance(
+        &self,
+        address: AccountAddress,
+        coin_type: &str,
+    ) -> AptosResult<Response<u64>> {
+        let resp = self
+            .view(&ViewRequest {
+                function: "0x1::coin::balance".parse()?,
+                type_arguments: vec![coin_type.parse()?],
+                arguments: vec![serde_json::Value::String(address.to_hex_literal())],
+            }, None)
+            .await?;
+
+        resp.and_then(|result| {
+            if result.len() != 1 {
+                Err(anyhow!("Wrong data size returned: {:?}", result).into())
+            } else {
+                result[0].as_str()
+                    .ok_or_else(|| anyhow!("Wrong data type returned: {:?}", result).into())
+                    .and_then(|v| v.parse::<u64>().map_err(|_e| anyhow!("Couldn't parse returned value: {:?}", result).into()))
+            }
+        })
+    }
```

Comment:
I would replace the existing balance call, with the BCS version since it's all abstracted away.  Don't need two of these as view function functions, on top of the existing function.

Reply:
view might be more expensive, so wanted it to be explicit. 

but I can just rename from get_account_balance to view_account_balance and have all callers updated.